tools/tertiary-analysis/scater/scater-filter.xml

```diff
@@ -1,113 +1,107 @@
-
-	-o OUTPUT-OBJECT-FILE, --output-object-file=OUTPUT-OBJECT-FILE
-		File name in which to store serialized R object of type Seurat
-
-	-u OUTPUT-CELLSELECT-FILE, --output-cellselect-file=OUTPUT-CELLSELECT-FILE
-        File name in which to store a matrix showing results of applying individual cell selection criteria.
-
-	-v OUTPUT-FEATURESELECT-FILE, --output-featureselect-file=OUTPUT-FEATURESELECT-FILE
-		File name in which to store a matrix showing results of applying individual feature selection criteria
-
-	-h, --help
-		Show this help message and exit
-
-    ]]></help>
-    <expand macro="citations" />
+<tool id="scater-filter" name="Scater Filter" version="@TOOL_VERSION@+galaxy0">
+  <description>cells and genes based on pre-calculated stats and QC metrics</description>
+  <macros>
+    <import>scater_macros.xml</import>
+  </macros>
+  <expand macro="requirements" />
+  <command detect_errors="exit_code"><![CDATA[
+ln -s '$R_scater_qc' R_scater_qc.rds &&
+
+scater-filter.R -i R_scater_qc.rds -o '$R_scater_filtered'
+#if $cell_parameters
+    #set cell_pars = ','.join([str($p['name']) for $p in $cell_parameters])
+    -s '${cell_pars}'
+    #set cell_mins = ','.join([str($p['min']) for $p in $cell_parameters])
+    -l '${cell_mins}'
+    #set maxs = ','.join([str($p['max']) for $p in $cell_parameters])
+    -j '${cell_maxs}'
+#end if
+#if $feature_parameters
+    #set ft_pars = ','.join([str($p['name']) for $p in $feature_parameters])
+    -t '${ft_pars}'
+    #set ft_mins = ','.join([str($p['min']) for $p in $feature_parameters])
+    -m '${ft_mins}'
+    #set ft_maxs = ','.join([str($p['max']) for $p in $feature_parameters])
+    -n '${ft_maxs}'
+#end if
+#if $cells_use:
+    -c '$cells_use'
+#end if
+#if $cells_discard:
+    -C '$cells_discard'
+#end if
+#if $features_use:
+    -f '$features_use'
+#end if
+#if $output_selected_cells:
+    -v '$selected_cells'
+#end if
+#if $output_selected_features:
+    -v '$selected_featuers'
+#end if
+
+  ]]></command>
+
+  <inputs>
+    <param name="R_scater_qc" type="data" format="rdata"
+           help="A serialized SingleCellExperiment object file in RDS format."/>
+
+    <repeat name="cell_parameters" title="Parameters used to filter cells" min="0">
+      <param name="name" type="text" value="total_features_by_counts" label="Name of the parameter to filter on" help="for example: total_features_by_counts, total_counts">
+        <option value="total_features_by_counts">total_features_by_counts</option>
+        <option value="total_counts">total_counts</option>
+      </param>
+      <param name="min" type="float" value="-1e9" label="Min value"/>
+      <param name="max" type="float" value="1e9" label="Max value"/>
+    </repeat>
+
+    <repeat name="feature_parameters" title="Parameters used to filter features" min="0">
+      <param name="name" type="text" value="n_cells_by_counts" label="Name of the parameter to filter on" help="for example: n_cells_by_counts, total_counts">
```

Comment:
Does this allow multiple field selection? I think you had a much neater approach on Scanpy where you could add variables progressively.

Reply:
Yes, this creates a UI that allows users to add variables progressively.